src/pkg/services/m365/api/item_pager.go

```diff
@@ -61,27 +61,56 @@ func (e EmptyDeltaLinker[T]) GetValue() []T {
 }
 
 // ---------------------------------------------------------------------------
-// generic handler for paging item ids in a container
+// generic handler for paging complete items in a container
 // ---------------------------------------------------------------------------
 
-type itemPager interface {
+type itemPager[T any] interface {
 	// getPage get a page with the specified options from graph
-	getPage(context.Context) (DeltaPageLinker, error)
+	getPage(context.Context) (PageLinker, error)
 	// setNext is used to pass in the next url got from graph
 	setNext(string)
-	// reset is used to clear delta url in delta pagers. When
-	// reset is called, we reset the state(delta url) that we
-	// currently have and start a new delta query without the token.
-	reset(context.Context)
 	// valuesIn gets us the values in a page
-	valuesIn(PageLinker) ([]getIDAndAddtler, error)
+	valuesIn(PageLinker) ([]T, error)
 }
 
-type getIDAndAddtler interface {
-	GetId() *string
-	GetAdditionalData() map[string]any
+func enumerateItems[T any](
+	ctx context.Context,
+	pager itemPager[T],
+) ([]T, error) {
+	var (
+		result   = make([]T, 0)
+		nextLink = "init"
```

Comment:
this needs a comment for why it's set to a non-empty value especially given the check below. I'm guessing it's to get the equivalent of do-while without lifting the first page fetch out of the loop?

Reply:
yup!  Will add a comment.